ultradns/resource_ultradns_tcpool.go

```diff
@@ -329,3 +334,20 @@ func makeSetFromRdata(rds []string, rdis []udnssdk.SBRDataInfo) *schema.Set {
 	}
 	return s
 }
+
+//State importer function resourceUltradnsTcpoolImport
+// State Function to seperate id into appropriate name and zone
+func resourceUltradnsTcpoolImport(
+	d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	newId := strings.TrimSuffix(d.Id(), ".")
+	attributes := strings.SplitN(newId, ":", 2)
+	if len(attributes) > 1 {
```

Comment:
for some of the resources we will need special handling for setting up the type for the resources, so i think it will be more manageable on resource basis and also recommended by terraform